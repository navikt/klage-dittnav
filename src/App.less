@import 'components/begrunnelse/begrunnelse.less';
@import 'components/summary/vedtak-summary.less';
@import 'pages/oppsummering-skjema-page/oppsummering-skjema-page.less';
@import 'media-queries.less';

/// LENKER ///
.lenke.no-background-style {
    background: none !important;
}

.lenke.no-background-style:focus {
    box-shadow: none;
    color: #0067c5;
}
