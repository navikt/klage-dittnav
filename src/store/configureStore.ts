import { applyMiddleware, createStore } from 'redux';
import thunkMiddleware from 'redux-thunk';
import reducer from './reducer';

export default function () {
    return createStore(reducer, applyMiddleware(thunkMiddleware));
}
