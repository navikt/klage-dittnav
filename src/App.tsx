import React, { useEffect, useState } from 'react';
import './App.css';
import { BrowserRouter as Router, Switch, Route } from 'react-router-dom';
import { routesConfig } from './utils/routes.config';
import Layout from './components/general/layout';
import { withRouter } from 'react-router-dom';
import queryString from 'query-string';
import { getTemaObject } from './services/klageService';
import { useDispatch } from 'react-redux';
import { setValgtYtelse } from './store/actions';
import NotFoundPage from './pages/not-found/not-found-page';
import NavFrontendSpinner from 'nav-frontend-spinner';
import { CenteredContainer } from './styled-components/main-styled-components';

const App = (props: any) => {
    const [loading, setLoading] = useState<boolean>(true);
    const [errorState, setErrorState] = useState<boolean>(false);
    const dispatch = useDispatch();

    useEffect(() => {
        if (props.location.search !== '') {
            const query = queryString.parse(props.location.search);
            if (query) {
                if (query.tema && !Array.isArray(query.tema)) {
                    getTemaObject(query.tema)
                        .then(res => {
                            let ytelse = '';
                            ytelse = query.ytelse && !Array.isArray(query.ytelse) ? query.ytelse : res.value;
                            dispatch(setValgtYtelse(ytelse));
                            setLoading(false);
                        })
                        .catch(err => {
                            if (err.response?.status === 404) {
                                setErrorState(true);
                                setLoading(false);
                                return;
                            }
                            console.log(err);
                        });
                } else {
                    setLoading(false);
                }
            } else {
                setLoading(false);
            }
        } else {
            setLoading(false);
        }
    }, [dispatch, props.location.search]);

    if (loading) {
        return (
            <CenteredContainer>
                <NavFrontendSpinner type={'XL'} />
            </CenteredContainer>
        );
    } else {
        if (errorState) {
            return <NotFoundPage />;
        } else {
            return (
                <Router>
                    <Layout>
                        <Switch>
                            {routesConfig.map(route => {
                                return <Route key={route.path} {...route} />;
                            })}
                        </Switch>
                    </Layout>
                </Router>
            );
        }
    }
};

export default withRouter(App);
