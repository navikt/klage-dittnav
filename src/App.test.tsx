import React from "react";
import renderer from "react-test-renderer";
import App from "./App";
import configureStore from "./store/configureStore";
import { Provider } from "react-redux";
import { BrowserRouter } from "react-router-dom";

describe("Welcome klage", () => {
  const store = configureStore();

  it("Renders correctly when there is no data", () => {
    const tree = renderer
      .create(
        <Provider store={store}>
          <BrowserRouter>
            <App />
          </BrowserRouter>
        </Provider>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
