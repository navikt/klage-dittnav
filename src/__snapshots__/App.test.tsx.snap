// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Welcome klage Renders correctly when there is no data 1`] = `
<div
  className="sc-AxjAm iJhWxX"
>
  <div
    className="sc-fzokOt sc-fzqBZW ftFonj"
  >
    <section>
      <div>
        <div
          className="sc-AxheI sc-fzoLsD jJmkkk"
        >
          <h1
            className="typo-sidetittel"
          >
            Klage eller anke p책 vedtak
          </h1>
        </div>
        <div
          className="sc-AxheI sc-fzoLsD jJmkkk"
        >
          <div
            className="nav-veilederpanel nav-veilederpanel--kompakt nav-veilederpanel--plakat"
          >
            <div
              className="nav-veileder nav-veileder--flytende"
            >
              <div
                className="nav-veileder__frame nav-veileder__frame--m"
              >
                <img
                  alt="Veileder"
                  src="Veileder.svg"
                />
              </div>
            </div>
            <div
              className="nav-veilederpanel__content"
            >
              <p
                className="typo-normal"
              >
                Hvis NAV har behandlet en sak som gjelder deg og du er uenig i vedtaket, har du flere valgmuligheter for 책 belyse saken bedre og f책 en ny vurdering. Start med 책 velge hvilket tema saken gjelder. Du finner denne informasjonen i vedtaket som du har mottatt fra NAV.
              </p>
            </div>
          </div>
        </div>
        <div
          className="sc-Axmtr sc-fzpans dcVwZu"
        >
          <h2
            className="typo-systemtittel"
          >
            Hvilket tema gjelder det?
          </h2>
        </div>
      </div>
      <div
        className="sc-fzoyAV sc-fznJRM bFbvvw"
      >
        <a
          className="lenkepanel lenkepanel-flex lenkepanel--border"
          href="/arbeid"
          onClick={[Function]}
        >
          <div>
            <h2
              className="typo-undertittel lenkepanel__heading"
            >
              Arbeid
            </h2>
            <p
              className="typo-normal"
            >
              Dagpenger, AAP, egen bedrift
            </p>
          </div>
          <span
            className="lenkepanel__indikator"
          />
        </a>
        <a
          className="lenkepanel lenkepanel-flex lenkepanel--border"
          href="/familie-og-barn"
          onClick={[Function]}
        >
          <div>
            <h2
              className="typo-undertittel lenkepanel__heading"
            >
              Familie og barn
            </h2>
            <p
              className="typo-normal"
            >
              Barnetrygd, foreldrepenger, pleie
            </p>
          </div>
          <span
            className="lenkepanel__indikator"
          />
        </a>
      </div>
    </section>
  </div>
</div>
`;
